harness/determined/deploy/aws/preflight.py

```diff
@@ -61,17 +61,24 @@ def fetch_instance_type_quota(boto_session: boto3.session.Session, quota_code: s
 # They can't be parsed by a simple yaml parser, but we can safely ignore them,
 # since we only make use of the default parameters section,
 # and it doesn't contain any such function calls.
-class LoaderIgnoreUnknown(yaml.SafeLoader):
+#
+# This strategy isn't a documented API in ruamel.yaml, but it was recommended by the ruamel.yaml
+# author in stack overflow[1].
+#
+# [1] https://stackoverflow.com/a/76870790
+class IgnoreUnknownTagConstructor(yaml.SafeConstructor):
     def ignore_unknown(self, node: Any) -> None:
         return None
 
 
-LoaderIgnoreUnknown.add_constructor(None, LoaderIgnoreUnknown.ignore_unknown)
+IgnoreUnknownTagConstructor.add_constructor(None, IgnoreUnknownTagConstructor.ignore_unknown)
 
 
 def get_default_cf_parameter(deployment_object: DeterminedDeployment, parameter: str) -> Any:
+    y = yaml.YAML(typ="safe", pure=True)
+    y.Constructor = IgnoreUnknownTagConstructor
     with open(deployment_object.template_path) as fin:
-        data = yaml.load(fin, Loader=LoaderIgnoreUnknown)
+        data = y.load(fin)
```

Comment:
@ioga please specifically review this section

Reply:
makes sense to me. I tested it and it seems to be working okay.